### PR TITLE
Subtitle CDN failover

### PR DESCRIPTION
**Pull request by jamierbower**
📺 What

Introduce CDN failover capability for subtitles.

> Tickets: [IPLAYERTVV1-11794](https://jira.dev.bbc.co.uk/browse/IPLAYERTVV1-11794)

🛠 How

- initialise BigscreenPlayer with a captions array.
- failover to subsequent sources if loading the subtitles xml is unsuccessful.

✅ Testing 

| Test engineer sign off | :x: |
| ---- | ---- |

 ✅ Acceptance criteria [Optional]

- [x] Subtitles continue to display if a cdn fails
- [x] Subtitles failover for VOD
- [x] Subtitles failover for Simulcast
- [x] Video failover not affected
- [x] Subtitles remain in sync after failover
- [x] Legacy and IMSC subtitles to have failover functionality

### CODE REVIEW

File: script/bigscreenplayer.js [hunk 1]
```diff
@@ -132,11 +132,10 @@ define('bigscreenplayer/bigscreenplayer',
 
         subtitles = Subtitles(
           playerComponent,
-          bigscreenPlayerData.media.captions || { captionsUrl: bigscreenPlayerData.media.captionsUrl },
```
Comment (jlks) [inline script/bigscreenplayer.js:135]: Can't you maintain backwards compatibility with the old API? This shouldn't be too hard to do

Comment (jamierbower) [reply script/bigscreenplayer.js:135]: We could, I'm not sure we need to as media selector will always be returning a captions object now

File: development/index.js [hunk 2]
```diff
@@ -15,20 +15,18 @@ require (['bigscreenplayer/bigscreenplayer'], function(BigscreenPlayer){
...
-          {
-            // Content from DASH IF testing assests (used in their reference player)
-            // https://reference.dashif.org/dash.js/v2.9.2/samples/dash-if-reference-player/index.htm
-            url: 'https://dash.akamaized.net/akamai/bbb_30fps/bbb_30fps.mpd',
-            cdn: 'dash.akamaized.net'
-          }
-        ]
-      }
-    };
+      captions: [],
```
Comment (domjacks) [inline development/index.js:18]: Maybe not doable, but is there a supplied example vid that _does_ have subtitles we could use?

Comment (jamierbower) [reply development/index.js:18]: I'll look into it!

**Review by matt-stephenson — APPROVED** [review]

**Pull request merged**